scripts/audit-allow-list

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env node
+
+/*
+This script takes the output of npm audit --json from stdin
+and writes a filtered version to stdout.
+The filtered version will have the entries listed in `AUDIT_ALLOW_LIST` removed.
+Specifically, each property of `vulnerabilities` in the input is matched by name in the allow list.
+
+Sample output of `npm audit --json`:
+{
+  // Other properties ...
+  "vulnerabilities": {
+    "meow": {
+      // Other properties ...
+    },
+    "trim-newlines": {
+      // Other properties ...
+    }
+  }
+}
+*/
+
+'use strict'
+const fs = require('fs');
+
+const USAGE = "Usage: npm audit --json | scripts/audit-allow-list"
+
+const AUDIT_ALLOW_LIST = [
+  {
+    name: "meow",
+    advisory: "https://www.npmjs.com/advisories/1753",
+    justification: "dependency of lerna (dev only)"
+  },
+  {
+    name: "trim-newlines",
+    advisory: "https://www.npmjs.com/advisories/1753",
+    justification: "dependency of lerna (dev only)"
+  }
+]
+
+/**
+ * @param audits - JavaScript object matching the schema of `npm audit --json`
+ * @param allowedAudits - List of package names to exclude from the audit
+*/
+function filterAudits(audits, allowedAudits) {
+  for (const vuln in audits.vulnerabilities) {
+    if (allowedAudits.includes(vuln)) {
+      // Skip error handling; we shouldn't have any non-configurable properties here.
+      delete audits.vulnerabilities[vuln]
+    }
+  }
+  return audits
+}
+
+const input = fs.readFileSync("/dev/stdin", "utf-8")
+if (input === "") {
+  console.error(USAGE)
+  process.exit(1)
+}
+
+// This function assumes `audits` has the right structure.
+// Just let the error terminate the process if the input doesn't match the schema.
+const filteredAudits = filterAudits(JSON.parse(input), AUDIT_ALLOW_LIST.map(x => x.name))
+
+// This is to mimic the behavior of `npm audit`:
+const numVulnerabilities = Object.keys(filteredAudits.vulnerabilities).length
+if (numVulnerabilities > 0) {
+  console.log(filteredAudits)
+  console.log("There are some unrecognized vulnerabilities from `npm audit`.")
+  console.log("Run `npm audit` to get pretty-printed output.")
```

Comment:
One thing we are losing here is the pretty-printed table of output in the run if it fails.  We could just invoke `npm audit` here if we prefer.  However, some downsides to that is it will be the full (unfiltered) output and it won't have `--audit-level=moderate` unless we want to hardcode that.

Reply:
Oh, I think it's `filteredAudits.vulnerabilities` that's null.  I tested on `npm` v7 locally and CI runs with `npm` v6.  The output of `npm audit` changed for v7.

Reply:
Removing this script so no longer applies